README.md

```diff
@@ -53,6 +53,40 @@ There are some mandatory fields, each marked with "MANDATORY" in a comment.
 
 See [monitoring.md](docs/monitoring.md).
 
+### Attribute redaction
+
+While Kubernetes `v1.Secrets` are designed to separate secret fields into their
+own types, referenced in other types' fields such as container environment
+variable `secretKeyRefs`, some users may wish to prevent certain fields on
+certain types from being sent to Snyk, while still scanning other fields on that
+type.
+
+The scanner supports this sort of redaction, as an optional array of paths
+alongside group-version-kind scan config:
+
+```yaml
+- apiGroups: ["apps"]
+    versions: ["*"]
+    resources:
+      - replicasets
+      - daemonsets
+      - deployments
+      - statefulsets
+    redactions:
+      - "spec.template.spec.containers.env"
+      - "metadata.managedFields"
+```
+
+These paths are dot-separated address for nested values, in the same format as
+arguments to `kubectl explain`. For example, the expression
+"spec.containers.env" will cause Kubernetes Pod container environment variables
+to be redacted. "containers" is an array, and implicitly each element of this
+array is redacted.
```

Comment:
Before hand-rolling an implementation, I explored jsonpath, jmespath, and jsonpatch libraries. The first two do not have spec (or implementation) support for subtree removals, and in practice any implementation that made use of jmespath syntax would have needed to implement the subset we choose to support - since we'll have to locate the node in the tree ourselves. I think this effectively rules them out, in favour of a more explicitly restricted, and therefore simple, syntax.

jsonpatch was looking good, but unfortunately does not support our first use case: removing subtrees under _all_ elements of al array, e.g. "redact all pod environment variables, from all containers" (pod.spec.containers.env, in the `kubectl explain`-ish syntax I ended up choosing). This is proposed here: https://github.com/json-patch/json-patch2/issues/18.

We could possibly use jsonpatch if we require users to specify exact json addresses to redact, e.g. individual environment variables (staying on that use-case).

WDYT?

Reply:
note to self: grammar error on line 80

Reply:
> jsonpath, jmespath, and jsonpatch libraries

Yeah, I have also done some look around long ago, they all have shortcomings: https://www.notion.so/snyk/Kubernetes-Monitor-Extending-to-Collect-CRDs-d278d6e900c1481b9d23d79fd3d52ee6?pvs=4#29939400a02a4c39a687b2e675e865fc

It seems to me simple handrolled implementation will be good enough
If we want to extend more, maybe we can take a look at `jq` in future: https://github.com/itchyny/gojq